devito/ir/clusters/cluster.py

```diff
@@ -343,6 +343,21 @@ def dspace(self):
         intervals = intervals.promote(lambda d: not d.is_Sub)
         intervals = intervals.zero(set(intervals.dimensions) - oobs)
 
+        # Intersect with intervals from buffered dimensions. Unions of
+        # buffered dimension intervals may result in shrinking time size
```

Comment:
"shrinking" feels odd here, since after all `time_M` will get a larger default

by shrinking here probably you mean that perhaps [0,1] becomes [0,0]

Reply:
The shrinking is introduced line 298, wouldn't it be cleaner to fix it there with slightly more checks

Reply:
maybe, not sure, haven't thought about this

part of the exercise, I guess